src/main/java/course_project/Main.java

```diff
@@ -0,0 +1,8 @@
+package course_project;
+
+public class Main {
+
+    public static void main(String[] args) {
+        new SeaBattle().run();
```

Comment:
When the opponent hits, I don't see him shooting again. So doesn't matter does he hit or miss, next turn is mine

Reply:
Manual ship placement is comfortable, also automatic is good

Reply:
Nice colour, testing is visual

Reply:
Well done! One bug to fix, except that - looks good! I like the nice visual representation, quite comfortable interacting with the console. Also, good abstractions, where needed. And good knowledge of Java Core functionality and Java libraries. 

To make better: 
Bug: When the opponent hits, I don't see him shooting again. So doesn't matter does he hit or miss, the next turn is mine
Abstractions are mostly separated by function, but usually, they are separated first by structure (model, service, printer, util, dao, etc). And so models don't usually have logic. 
Better use methods with return values, void methods are not thread-safe and difficult to test